### DISCUSSION TITLE
uniswap.make_trade_output with eth output renders a wETH rather than ETH

### DISCUSSION THREAD

**Discussion by ToposFinance** [uniswap-python/uniswap-python#122] [General]
uniswap.make_trade_output(usdt, eth, 1 * 10 ** 18, recipient)  gives wETH in my test.  perhaps it doesn't matter.

**Comment by ToposFinance** [comment]
actually, it definitely matters....

**Comment by liquid-8** [comment]
Well, both `exactInputSingle()` and `exactOutputSingle()` methods act this way if you're talking about version=3. At the moment, `make_trade()` and `make_trade_output()` wrap them. If you need ETH exactly, you should call manually smth like
`uniswap.router.functions.unwrapWETH9(qty, uniswap.address).call()`
after successful swap. To make that automatically `uniV3router.multicall()` should be wrapped.
